### PR TITLE
docker container as executable

### PR DESCRIPTION
**Pull request by chrishah**
Pull request as discussed with Sheila on the GATK mailing list.

Made small changes to Dockerfile and docker_helper.sh to enable the usage of the Picard docker container as an executable. With the image built from this context (`docker build -t custompicard .`) I can run, e.g.:
`docker run -i -t -v $(pwd):/usr/working custompicard MergeSamFiles I=first.bam I=second.bam O=merged.bam` 
successfully if the bam files are present in the current working directory.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,4 +28,8 @@ RUN ant clean all && \
     rm -rf lib && \
     rm build.xml
 
-ENTRYPOINT ["./docker_helper.sh"]
\ No newline at end of file
+RUN mkdir /usr/working
+WORKDIR /usr/working
```
Comment (bradtaylor) [inline Dockerfile]: This was a good catch!

File: Dockerfile [hunk 2]
```diff
@@ -28,4 +28,8 @@ RUN ant clean all && \
     rm -rf lib && \
     rm build.xml
 
-ENTRYPOINT ["./docker_helper.sh"]
\ No newline at end of file
+RUN mkdir /usr/working
```
Comment (bradtaylor) [inline Dockerfile]: Should this mkdir be moved to a previous run command in order to avoid an additional layer, and for more efficient caching?

**Comment by bradtaylor** [comment]
Thank you for this PR!  And for your interest in Docker, which is new to picard. A couple little comments/questions, otherwise it looks great.

One thing I do notice is that, looking at the example docker run command you supplied, I don't believe the -i and -t flags are necessary. My understanding is they are used for interactive sessions, as when running /bin/bash. For example, I was able to successfully execute the following command:
```docker run -v $(pwd):/usr/working custompicard SamToFastq I=ubams_testing/newnewnew.bam F=test.fastq F2=test2.fastq```

One other thing to note:
The ENTRYPOINT command does not play nice with some systems we are building internally. So we have an agenda item to remove it. At that point, you would specify the full java invocation as an argument to the container run (the helper script would go away). In place of the entrypoint command will probably be a CMD simply echoing a usage message with instructions on composing an appropriate run command. This will likely happen soon. In the interim, I will merge this PR once I here back from you regarding the questions I mentioned.

We here at Broad really appreciate your interest in Picard. Thanks for your contribution!

**Comment by chrishah** [comment]
Hi Brad,
Glad I could contribute something useful!
I agree with you, the `-i` ant `-t` flags in my command suggestion are not actually necessary. 
The `CMD [""]` is a solution for the behaviour discussed a bit in [this](https://github.com/docker/docker/issues/3753) Docker issue - without it I got an error message "/bin/bash is not a valid command" when running the container without any arguments. I hadn't thought about efficient caching at all, to be honest.
Sure, I've read elsewhere that `ENTRYPOINT` can be problematic in some cases. Looking forward to seeing your solution then! 
Thanks!
Christoph

**Comment by bradtaylor** [comment]
Only now seeing Christoph's response. Sorry for the delay!  Going to merge as-is.

**Pull request merged**